channel/state/state.go

```diff
@@ -231,8 +237,8 @@ func StateFromFixedAndVariablePart(f FixedPart, v VariablePart) State {
 		ChainId:           f.ChainId,
 		Participants:      f.Participants,
 		ChannelNonce:      f.ChannelNonce,
-		AppDefinition:     f.AppDefinition,
-		ChallengeDuration: f.ChallengeDuration,
+		ChallengeDuration: v.ChallengeDuration,
+		AppDefinition:     v.AppDefinition,
```

Comment:
out of curiosity, is the order swap here intentional?

Reply:
I don't think so!

Reply:
To help me understand: the re-ordering in this case is not meaningful, right?

Reply:
I think the ordering is somewhat meaningful: in Go you can construct literal struct instances without naming the fields (you just pass in the arguments in a certain order). So the change could in principle break some other bits of code.